lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/DefaultNSMessageCodec.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.network.impl;
+
+import org.apache.reef.wake.Identifier;
+import org.apache.reef.wake.IdentifierFactory;
+import org.apache.reef.wake.remote.Codec;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * NSMessage codec.
+ *
+ */
+final class DefaultNSMessageCodec implements Codec<DefaultNSMessage> {
+
+  private final IdentifierFactory factory;
+  private final Map<String, NSConnectionFactory> connectionFactoryMap;
+  private final Map<String, Boolean> isStreamingCodecMap;
+
+  /**
+   * Constructs a network message codec.
+   */
+  DefaultNSMessageCodec(
+      final IdentifierFactory factory,
+      final Map<String, NSConnectionFactory> connectionFactoryMap,
+      final Map<String, Boolean> isStreamingCodecMap) {
```

Comment:
Don't like to maintain a separate map for each data member. 
Why don't we maintain a map that has <String, DataX> where DataX has all the relevant members?